src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionProxy.java

```diff
@@ -169,7 +169,7 @@ public HostInfo getCurrentHostInfo() {
   @Override
   public void setCurrentConnection(JdbcConnection connection, HostInfo info) {
     try {
-      if (this.currentConnection != null && !this.currentConnection.isClosed()) {
+      if (this.currentConnection != null && !this.currentConnection.equals(connection)  && !this.currentConnection.isClosed()) {
```

Comment:
Not sure i understand. We close the current connection if we are setting a new connection as current?

Reply:
Yup, thats right. Its cleaning up the "old" current connection before we assign a new current connection. 
This fix here makes sure that if the new and old one the same, dont close it.